velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -914,6 +924,11 @@ bool assertEqualResults(
     const std::vector<RowVectorPtr>& expected,
     const std::vector<RowVectorPtr>& actual) {
   MaterializedRowMultiset expectedRows;
+  VELOX_CHECK_GT(expected.size(), 0);
+  const auto type = expected.at(0)->childAt(0)->type();
+  // Confirm that the expected and actual types match
+  // VELOX_CHECK_EQ(expected.at(0)->childAt(0)->type(),
```

Comment:
We are comparing the pointers here and not the types.
I also see that there is a type check below on line ~1003 `if (!equalTypeKinds(*expectedRows.begin(), *actualRows.begin())) {`
In that function, can you change this check to `if (!expected.at(0)->type()->equivalent(*(actual.at(0)->type()))) {`
Also, looking at the code in the function below,  we should expect `expected` to be empty.
Set expectedType to `nullptr` if `expectedRows` is empty below, else infer the type from the first element as it is now.
In the call below, if expectedType is nullptr, then try to get the type from `actual`.
Please add tests for these cases if not already there.

Reply:
Correction: I think we need to pass both actualType and expectedType to the main `assertEqualResults` where the rows and types are compared.

Reply:
Fixed.